libyul/AsmAnalysis.cpp

```diff
@@ -167,6 +167,26 @@ void AsmAnalyzer::operator()(Assignment const& _assignment)
 	size_t const numVariables = _assignment.variableNames.size();
 	yulAssert(numVariables >= 1, "");
 
+	map<YulString, vector<Identifier const*>> variables;
+	for (auto const& _variableName: _assignment.variableNames)
+		variables[_variableName.name].emplace_back(&_variableName);
+
+	for (auto const& [name, occurrences]: variables)
+		if (occurrences.size() > 1)
+		{
+			SecondarySourceLocation ssl;
+			for (auto const* occurrence: occurrences)
+				ssl.append("Variable occurrence", occurrence->location);
```

Comment:
The secondary locations might actually be overkill...

Reply:
Yeah, also crossed my mind.

Reply:
Removed them.